### PR TITLE
demo: update demo

### LINKED ISSUE #44863: Time-picker 组件 clearText 属性不生效
### Reproduction link

[https://stackblitz.com/edit/react-gbifab?file=demo.tsx](https://stackblitz.com/edit/react-gbifab?file=demo.tsx)

### Steps to reproduce

选择一个时间，hover 在清除按钮上

### What is expected?

显示 clearText 文案

### What is actually happening?

未显示 clearText 文案

| Environment | Info |
| --- | --- |
| antd | 5.9.0 |
| React | latest |
| System | MacOS 13.5.1 (22G90) |
| Browser | Google Chrome 版本 117.0.5938.62（正式版本） (arm64) |

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- close #44863

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | demo: update demo |
| 🇨🇳 Chinese |          - |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at dbb18d0</samp>

This pull request refactors the date picker demo code, exports the date picker props type, and removes the deprecated clearText prop from the time picker documentation. These changes improve the code quality, type safety, and user experience of the components.

### <samp>🤖 Generated by Copilot at dbb18d0</samp>

*  Refactor `cellRender` function and `style` object in date picker demo to improve readability and maintainability ([link](https://github.com/ant-design/ant-design/pull/44864/files?diff=unified&w=0#diff-7bef207684561bc2802ba1dde4264ee5a219524db1d1a45ff643a2579948c15eL3-R27))
* Export `DatePickerProps` type from `antd` library to use in `cellRender` function type annotation ([link](https://github.com/ant-design/ant-design/pull/44864/files?diff=unified&w=0#diff-78eb7aa1662e48dc14665cbcead280795a60503742024dd031b87311186be5eeR3))
* Remove deprecated `clearText` prop from `TimePicker` component and documentation ([link](https://github.com/ant-design/ant-design/pull/44864/files?diff=unified&w=0#diff-93e8155915622d7eec5ef5f00cc60b8a700baed676ab713aca37c6281a82c310L59), [link](https://github.com/ant-design/ant-design/pull/44864/files?diff=unified&w=0#diff-2dda2de66ee5e0215c478323e414235aa4613c4d93674510732ef1d043508061L59))

### CODE REVIEW

**Review by afc163 — APPROVED** [review]

File: components/date-picker/demo/cell-render.tsx [hunk 1]
```diff
@@ -1,42 +1,30 @@
 import React from 'react';
 import { DatePicker, Space } from 'antd';
+import type { DatePickerProps } from 'antd';
 
-const { RangePicker } = DatePicker;
+const style: React.CSSProperties = {
+  borderRadius: '50%',
+  border: '1px solid #1677ff',
```
Comment (afc163) [inline components/date-picker/demo/cell-render.tsx:7]: 用 token

File: components/date-picker/demo/cell-render.tsx [hunk 2]
```diff
@@ -1,42 +1,30 @@
...
+};
+
+const cellRender: DatePickerProps['cellRender'] = (current, info) => {
+  if (info.type !== 'date') {
+    return info.originNode;
+  }
+  if (typeof current === 'number') {
+    return <div className="ant-picker-cell-inner">{current}</div>;
```
Comment (afc163) [inline components/date-picker/demo/cell-render.tsx:15]: ant-picker-cell-inner 这个样式现在还有用么？在没 hash 的情况下

Comment (li-jia-nan) [reply components/date-picker/demo/cell-render.tsx:15]: 有用的，样式里面有定义到：

```ts
pickerCellInnerCls: `${componentCls}-cell-inner`,
```

File: components/date-picker/demo/cell-render.tsx (hunk 1 above)
Comment (li-jia-nan) [reply components/date-picker/demo/cell-render.tsx:7]: 我改一下

Comment (li-jia-nan) [reply components/date-picker/demo/cell-render.tsx:6]: 用 token 的话，这部分还是得写到组件里面，如果要定义在外面的话，需要用参数把 token 传进去，但是感觉 demo 没必要写的太复杂……这个PR本身就是为了做简化

Comment (MadCcc) [reply components/date-picker/demo/cell-render.tsx:6]: 要考虑下主题切换

Comment (li-jia-nan) [reply components/date-picker/demo/cell-render.tsx]: 改成 token 了

**Pull request merged**